docs/cqrs/local_execution/index.md

```diff
@@ -0,0 +1,104 @@
+
+## Theory
+
+Being able to call CQRS objects locally without going through HTTP pipeline is achieved by using a separate ASP.NET Core pipeline that is used with mocked [HttpContext]. This enables us to re-use all the middlewares created for normal HTTP execution (which applies to all middlewares provided by the CoreLibrary) without changes. This option also enables using the same handlers as the normal executions.
+
+Nevertheless, the local execution only mimics the HTTP pipeline. Local execution does not involve proper request/response, thus it comes with limitations:
+
+1. [HttpContext.Connection], [HttpContext.WebSockets], [HttpContext.Session] are represented by empty objects - report null/empty data and all actions either throw or are no-op,
+2. [HttpContext.User] is passed by the caller (and can, but does not have to be, real),
+3. [HttpContext.Response] is an empty object, meaning that it ignores any writes to it (both body and headers will be lost),
+4. [HttpContext.Request] does not have any body, nor other HTTP metadata like path, method or so. The only thing that is implemented are headers, which can be passed when calling local execution.
+5. The features provided by [DefaultHttpContext] are only partly available and we don't guarantee any feature to be available.
+
+We found that, albeit some features that rely on HTTP will be ignored, most of the middlewares will be fully working.
+
+Local executors preserve the semantics of HTTP calls, meaning that:
+
+1. They are run in a separate DI scope,
+2. They are stateless, and don't share anything with the parent call.
+
+## Packages
+
+| Package                  | Link        | Application in section |
+| ------------------------ | ----------- | ---------------------- |
+| LeanCode.CQRS.AspNetCore | [![NuGet version (LeanCode.CQRS.AspNetCore)](https://img.shields.io/nuget/vpre/LeanCode.CQRS.AspNetCore.svg?style=flat-square&logo=nuget)](https://www.nuget.org/packages/LeanCode.CQRS.AspNetCore) | Configuration |
+
+## Configuration
+
+To use local execution, you have to explicitly register local executors (query/command/operation separately). This can be done by chaining calls to [AddCQRS(...)], specifying pipelines for local execution. For example:
+
+```csharp
+public override void ConfigureServices(IServiceCollection services)
+{
+    services
+        .AddCQRS(TypesCatalog.Of<ExampleCommand>(), TypesCatalog.Of<ExampleHandler>())
+        .WithLocalCommands(c => c.Secure().Validate().TranslateExceptions().CommitTransaction<CoreDbContext>())
+        .WithLocalQueries(c => c.Validate().TranslateExceptions())
+        .WithLocalOperations(c => c.Secure().TranslateExceptions().CommitTransaction<CoreDbContext>());
+}
+```
```

Comment:
Nit: We could format this so that we do not need to scroll horizontally on read the docs.

Reply:
Good point, will change.